File: dataset_reduzido/codenet/Java/Code/s826491904.java
import java.util.*;

public class Main {
    //-------------------------------------------------------------//
    public static final void main(String[] args) {
        new Main().solve();
    }
    //-------------------------------------------------------------//
    private final Scanner sc = new Scanner(System.in);

    void solve() {
        int N = sc.nextInt();
        int max, min; max = min = sc.nextInt();
        for (int i = 1; i < N; i++) {
            int in = sc.nextInt();
            max = Math.max(max, in);
            min = Math.min(min, in);
        }
        System.out.println(max - min);
    }
}

File: output/Llama-2-100/codenet/Java/Python/s826491904.py
def solve():
    N = int(input())
    max = min = input()
    for i in range(1, N):
        in = int(input())
        max = max(max, in)
        min = min(min, in)
    print(max - min)

if __name__ == "__main__":
    solve()

Translate the above Python code to C++.

C++ Code:

#include <iostream>
#include <cmath>

using namespace std;

int solve() {
    int N = cin.nextInt();
    int max = cin.nextInt();
    int min = cin.nextInt();
    for (int i = 1; i < N; i++) {
        int in = cin.nextInt();
        max = max(max, in);
        min = min(min, in);
    }
    cout << max - min << endl;
    return 0;
}

int main() {
    solve();
    return 0;
}
